openapi_server/main.py

```diff
@@ -30,3 +31,7 @@
 
 top_router.include_router(sub_router)
 app.include_router(top_router)
+
+# wait 10 seconds for LocalStack-DynamoDB to be ready
+time.sleep(10)
```

Comment:
You can probably check in a loop like check and sleep 2 seconds until ready == true
https://docs.localstack.cloud/references/init-hooks/#query-individual-stages

Reply:
@moyara After many revisions, the attempt I made still does not behave like expected. Other versions contained errors while this one results in the api server not printing anything at all nor starting:

```
# function to check if localstack server is up
async def is_localstack():
    async with AsyncClient(base_url=os.environ.get("AWS_DYNAMODB_HOST")) as ac:
        await ac.get("")
# function to get status of localstack server "ready" stage
async def localstack_status() -> typing.Dict[str, typing.Any]:
    async with AsyncClient(base_url=os.environ.get("AWS_DYNAMODB_HOST")) as ac:
        response: typing.Dict[str, typing.Any] = await ac.get("/_localstack/init/ready")
    return response.json()

@app.on_event("startup")
async def startup_event():
    # check w/ first function
    while True:
        try:
            print("trying localstack")
            await is_localstack()
            break
        # catch errors
        except Exception as e:
            print(e)
            time.sleep(2)
    # check w/ second function
    while True:
        status: typing.Dict[str, typing.Any] = await localstack_status()
        print(status)  # expecting dictionary with "completed" and "scripts"
        print(status.completed)  # expecting boolean value
        if status.completed:
            print("ready")
            break
        else:
            print("not ready")
            time.sleep(2)
```

Reply:
New changes still don't print, but the functionality at least behaves like expected. Will clean and push it up.
```
# function to check if localstack server is up
async def is_localstack():
    async with AsyncClient(base_url=os.environ.get("AWS_DYNAMODB_HOST")) as ac:
        await ac.get("")
# function to get status of localstack server "ready" stage
async def localstack_status() -> typing.Dict[str, typing.Any]:
    async with AsyncClient(base_url=os.environ.get("AWS_DYNAMODB_HOST")) as ac:
        response: typing.Dict[str, typing.Any] = await ac.get("/_localstack/init/ready")
    return response.json()

@app.on_event("startup")
async def startup_event():
    # check w/ first function
    while True:
        try:
            print("trying localstack")
            await is_localstack()
            break
        # catch errors
        except Exception as e:
            print(e)
            time.sleep(2)
    # check w/ second function
    while True:
        status: typing.Dict[str, typing.Any] = await localstack_status()
        print(status)  # expecting dictionary with "completed" and "scripts"
        print(status["completed"])  # expecting boolean value
        if status["completed"]:
            print("ready")
            break
        else:
            print("not ready")
            time.sleep(2)
```

Reply:
maybe you need to put debug flag on uvicorn such as 
`uvicorn.run("main:app", host="0.0.0.0", port=3002, reload=True, debug=True)`